tests/FakeItEasy.IntegrationTests/FakingClassesTests.cs

```diff
@@ -12,10 +11,13 @@ public class FakingClassesTests
         [Test]
         public void Should_be_able_to_get_a_fake_value_of_uri_type()
         {
-            using (Fake.CreateScope(new NullFakeObjectContainer()))
-            {
-                A.Fake<Uri>();
-            }
+            // Arrange
+
+            // Act
+            var fake = A.Fake<Uri>();
+
+            // Assert
+            Fake.GetFakeManager(fake).Should().NotBeNull("because we should be able to create a fake Uri");

```

Comment:
I'm not sure about this.

Is there a possible code path for `A.Fake<T>()` to return `null`?

As I understood it, this test was added because there was some infinite recursion for certain constructors. I guess in the case of `System.Uri` the `(Uri, Uri)` constructor would have made it happen and a `StackOverflowException` would be thrown.

In that case we should really change this test to be `Should_be_able_to_get_a_fake_value_for_a_type_with_a_constructor_taking_arguments_of_that_type` and the assertion should revert to asserting that no exception is thrown. We can also switch to a custom test type instead of using `System.Uri` to avoid misleading people again.